### PR TITLE
.Net: Assistant chat enhancement

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Explore enhanced assistant chat interactions.

### Description

- Added `Example72_AssistantCollaboration `and `Example73_AssistantAuthoring`
- Refined assistant and plugin calling patterns
- Simplified kernel-examples (esp cleanup)
- Ordered kernel-examples

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/Example72_AssistantCollaboration.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            copyWriter = await CreateCopyWriterAsync();
+            // Create art-director assistant to review ideas, provide feedback and final approval
+            artDirector = await CreateArtDirectorAsync();
+            // Create coordinator assistant to oversee collaboration
+            coordinator =
+                await new AssistantBuilder()
+                    .WithOpenAIChatCompletion(OpenAIFunctionEnabledModel, TestConfiguration.OpenAI.ApiKey)
+                    .WithInstructions("Coordinate the repeated interaction between a copy-writer and an art-director on a single thread until the art-director approves the copy.  First call the copywriter, then have the art-director review the copy.  Respond with the copy when approved by the art-director and summarize why it is good.")
+                    .WithPlugin(copyWriter.AsPlugin())
+                    .WithPlugin(artDirector.AsPlugin())
```
[the remark this answers is not in the document]
Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/Example72_AssistantCollaboration.cs:114]: It is: critical...but this approach is likely a dead-end as it isn't consistant across models.  I do like defering to the model, but one aspect of this approach I dislike is that the internal chat isn't surfaced without extra work (just like all of our function calling).

Anyway, this is all very fluid right now.

Wanted to explore this before investing in code-based coordination strategies.

I'm currently working to surface data-points to inform execution in this space and wanted to maintain visibility.

File: dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs [hunk 2]
```diff
@@ -118,34 +117,25 @@ private static async Task RunAsFunctionAsync()
...
+            // Display result.
+            Console.WriteLine(response ?? $"No response from assistant: {assistant.Id}");
         }
         finally
         {
-            await Task.WhenAll(
-                assistant.DeleteThreadAsync(threadId),
-                assistant.DeleteAsync());
+            // Clean-up (storage costs $)
+            await assistant.DeleteAsync();
```
Comment (glahaye) [inline dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs:138]: Not deleting the thread anymore?

Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs:138]: Its being deleted automatically now

File: dotnet/samples/KernelSyntaxExamples/Example72_AssistantCollaboration.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    /// Specific model is required that supports assistants and function calling.
+    /// Currently this is limited to Open AI hosted services.
+    /// </summary>
+    private const string OpenAIFunctionEnabledModel = "gpt-4-0613";
+
+    // Track assistants for clean-up
+    private static readonly List<IAssistant> s_assistants = new();
+
+    /// <summary>
+    /// Show how to combine coordinate multiple assistants.
```
Comment (glahaye) [inline dotnet/samples/KernelSyntaxExamples/Example72_AssistantCollaboration.cs:25]:
```suggestion
    /// Show how to combine and coordinate multiple assistants.
```

**Review by matthewbolanos — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/Program.cs [hunk 4]
```diff
@@ -21,7 +21,7 @@ public static async Task Main(string[] args)
         using CancellationTokenSource cancellationTokenSource = new();
         CancellationToken cancelToken = cancellationTokenSource.ConsoleCancellationToken();
 
-        string? defaultFilter = null; // Modify to filter examples
+        string? defaultFilter = "Example73_AssistantAuthoring"; // Modify to filter examples
```
Comment (glahaye) [inline dotnet/samples/KernelSyntaxExamples/Program.cs:24]: You don't want to check this in

**Review by glahaye — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/Example69_Pipelining.cs [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+
+<<<<<<<< HEAD:dotnet/samples/KernelSyntaxExamples/Example69_Pipelining.cs
```
Comment (glahaye) [inline dotnet/samples/KernelSyntaxExamples/Example69_Pipelining.cs:13]: Better fix that merge!  And the one in the other file too...

**Review by alliscode — APPROVED** [review]

**Pull request merged**